### PR TITLE
Improve bastion section

### PR DESCRIPTION
**Pull request by hebelsan**
**How to categorize this PR?**

/area usability
/kind enhancement

**What this PR does / why we need it**:
This PR improves the usability of the bastion section introduced with https://github.com/gardener/gardener/pull/10233 by exposing more fields as optional.
The bastion controllers will handle the missing fields with a uniform approach.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Comment by hebelsan** [comment]
@rfranzke We also need some validation for the bastion fields.
I would check if the passed bastionImage and bastionMachineType exists in the CloudProfile.
Furthermore I would check if they are compatible - meaning the architecture of the machineType and image match.
Since this is not provider-specific it belong into gardener/gardener, right?
I would work on the validation in a separate PR.

**Comment by rfranzke** [comment]
> Since this is not provider-specific it belong into gardener/gardener, right?

Yes, sounds reasonable 👍🏻

File: pkg/apis/core/v1beta1/types_cloudprofile.go [hunk 1]
```diff
@@ -225,17 +225,17 @@ type VolumeType struct {
 // Bastion contains the bastions creation info
 type Bastion struct {
 	// MachineImage contains the bastions machine image properties
-	MachineImage BastionMachineImage `json:"machineImage" protobuf:"bytes,1,name=machineImage"`
+	MachineImage *BastionMachineImage `json:"machineImage,omitempty" protobuf:"bytes,1,opt,name=machineImage"`
```
Comment (rfranzke) [inline pkg/apis/core/v1beta1/types_cloudprofile.go:229]: `// +optional` tag missing

File: pkg/apis/core/v1beta1/types_cloudprofile.go [hunk 2]
```diff
@@ -225,17 +225,17 @@ type VolumeType struct {
...
 	// MachineType contains the bastions machine type properties
-	MachineType BastionMachineType `json:"machineType" protobuf:"bytes,2,name=machineType"`
+	MachineType *BastionMachineType `json:"machineType,omitempty" protobuf:"bytes,2,opt,name=machineType"`
```
Comment (rfranzke) [inline pkg/apis/core/v1beta1/types_cloudprofile.go:232]: same here

File: pkg/apis/core/v1beta1/types_cloudprofile.go [hunk 3]
```diff
@@ -225,17 +225,17 @@ type VolumeType struct {
...
 }
 
 // BastionMachineImage contains the bastions machine image properties
 type BastionMachineImage struct {
 	// Name of the machine image
 	Name string `json:"name" protobuf:"bytes,1,name=name"`
 	// Version of the machine image
-	Version string `json:"version" protobuf:"bytes,2,name=version"`
+	Version *string `json:"version,omitempty" protobuf:"bytes,2,opt,name=version"`
```
Comment (rfranzke) [inline pkg/apis/core/v1beta1/types_cloudprofile.go:241]: same here

**Review by rfranzke — COMMENTED** [review]
/lgtm
/approve

**Comment by rfranzke** [comment]
@hebelsan Do you think the release note is understandable when somebody (other than you, @kon-angelo or @AndreasBurger) reads it? 😉 Please either refine it (so that even somebody can understand it who has not developed this feature or is familiar with this GEP), or remove it entirely :)

**Comment by hebelsan** [comment]
Removed the release note

**Pull request merged**